app/views/lectures/edit/_people.html.erb

```diff
@@ -19,96 +19,104 @@
          aria-labelledby="heading"
          data-parent="#lectureAccordion">
       <div class="card-body">
-        <%= form_with model: lecture, html: { id: 'lecture-form' } do |f| %>
-          <div class="form-row">
-            <div class="col-4 form-group">
-              <%= f.label :teacher_id,
-                          t('basics.teacher') %>
-              <%= helpdesk(t('admin.lecture.info.teacher'), false) %>
-              <div id="lecture_teacher_select" %>
-                <%= f.select :teacher_id,
-                             options_for_select([[lecture.teacher.info,
-                                                  lecture.teacher.id]],
-                                                  lecture.teacher.id),
-                             {},
-                             { class: 'selectize',
-                               data: { ajax: true,
-                                       model: 'user',
-                                       filled: false,
-                                       placeholder: t('basics.enter_two_letters'),
-                                       no_results: t('basics.no_results') } } %>
-                <div class="invalid-feedback" id="lecture-teacher-error">
+        <% if current_user.can_view_users?(lecture) %>
+          <%= form_with model: lecture, html: { id: 'lecture-form' } do |f| %>
+            <div class="form-row">
+              <div class="col-4 form-group">
+                <%= f.label :teacher_id,
+                            t('basics.teacher') %>
+                <%= helpdesk(t('admin.lecture.info.teacher'), false) %>
+                <div id="lecture_teacher_select" %>
+                  <%= f.select :teacher_id,
+                               options_for_select([[lecture.teacher.info,
+                                                    lecture.teacher.id]],
+                                                    lecture.teacher.id),
+                               {},
+                               { class: 'selectize',
+                                 data: { ajax: true,
+                                         model: 'user',
```

Comment:
I've seen this line in:
```
fill_path = Routes.fill_user_select_path();
```

in `app/assets/javascripts/_selectize_turbolink_fix.js`. But I can't figure out, how this `_people.html.erb` knows which method on the model `user` to call by just specifying `model: 'user'` here.

Reply:
This is all done in `app/assets/javascripts/_selectize_turbolink_fix.js`. In

`fill_path = Routes.fill_user_select_path();`

the route to the `fill_user_select` action in the users controller is provided in JS by the [js-routes](https://github.com/railsware/js-routes) gem. Then this route is given to TomSelect in 

```
        load: function(query, callback) {
          var url;
          if (send_data || !loaded) {
            url = fill_path + "?course_id=" + courseId + "&q=" + encodeURIComponent(query);
            fetch(url).then(function(response) {
              return response.json();
```
(see e.g. [here](https://tom-select.js.org/examples/remote/)).